pylearn2/models/mlp.py

```diff
@@ -2024,15 +2020,6 @@ def _modify_updates(self, updates):
             if W in updates:
                 updates[W] = updates[W] * self.mask
 
-        if self.max_row_norm is not None:
-            W, = self.transformer.get_params()
-            if W in updates:
-                updated_W = updates[W]
-                row_norms = T.sqrt(T.sum(T.sqr(updated_W), axis=1))
-                desired_norms = T.clip(row_norms, 0, self.max_row_norm)
-                scales = desired_norms / (1e-7 + row_norms)
-                updates[W] = updated_W * scales.dimshuffle(0, 'x')
-
         if self.max_col_norm is not None or self.min_col_norm is not None:
```

Comment:
yeah, let's integrate the minimum norm constraint.
And yes, you should forbid both row and column norm constraints at same time, that's what current code is doing now.